prusti-specs/src/specifications/preparser.rs

```diff
@@ -289,557 +58,547 @@ impl Parse for SpecEntArgs {
-        // TODO: (after discussion on syntax)
-        //   expr |= |arg, arg, ...| requires(...)
-        //   expr |= requires(...)
-        // by this point we have already consumed expr and |=
-
-        // parse args
-        let vars = if !self.input.check_and_consume_operator("|") {
-            vec![]
-        } else {
-            let token_stream = self.input.create_stream_until("|");
-            let all_args: SpecEntArgs = syn::parse2(token_stream)?;
-            if !self.input.check_and_consume_operator("|") {
-                return Err(self.error_expected_or());
-            }
+        let all_args: ForAllArgs = syn::parse2(arg_tokens)?;
+        if !self.consume_operator("|") {
+            return Err(self.error_expected("`|`"));
+        }
+        let vars: Vec<Arg> =
             all_args.args.into_iter()
                          .map(|var| Arg { typ: var.typ, name: var.name })
-                         .collect()
-        };
-
-        if let Some(group) = self.input.check_and_consume_block(Delimiter::Bracket) {
-            // parse specification
-            let mut parser = Parser::from_token_stream(group.stream());
-            let mut pres = vec![];
-            let mut posts = vec![];
-            // TODO: pledges
-            //let mut pledges = vec![];
-            loop {
-                use common::SpecType::*;
-                if parser.input.is_empty() {
-                    break
-                }
-                let kind = if parser.input.check_and_consume_keyword("requires") {
-                    Precondition
-                } else if parser.input.check_and_consume_keyword("ensures") {
-                    Postcondition
-                } else {
-                    // TODO: invariant, pledge ...
-                    return Err(self.error_expected_specification());
-                };
-                let conjunct = if let Some(group) = parser.input.check_and_consume_parenthesized_block() {
-                    let mut parser = Parser::from_token_stream(group.stream());
-                    parser.extract_assertion()?
-                } else {
-                    return Err(self.error_expected_parenthesis());
-                };
-                if !parser.input.check_and_consume_operator(",") {
-                    if !parser.input.is_empty() {
-                        return Err(self.error_expected_comma());
+                         .collect();
+
+        let body = self.parse_prusti()?;
+
+        let mut trigger_set = TriggerSet(vec![]);
+
+        if self.consume_operator(",") {
+            if !self.consume_keyword("triggers") {
+                return Err(self.error_expected("`triggers`"));
+            }
+            if !self.consume_operator("=") {
+                return Err(self.error_expected("`=`"));
+            }
+
+            let arr: Result<syn::ExprArray, Error> = syn::parse2(self.create_stream_remaining());
+            match arr {
+                Ok(arr) => {
```

Comment:
```suggestion
            let arr = syn::parse2(self.create_stream_remaining())
                .map_err(|err| self.error_expected_tuple(err.span()))?;
```
Maybe with some type-hints?

Reply:
Addressed by c77b6d0b1638c36b81eddb1e87af37c3f1fff212